py/server/tests/test_function_generated_table.py

```diff
@@ -111,6 +111,36 @@ def table_generator_function():
             self.assertEqual(result_str, 'test string')
             self.assertEqual(result_int, 12345)
 
+    def test_generated_table_kwargs(self):
+        def table_generator_function(nrows, query_string):
+            return empty_table(nrows).update(query_string)
+
+        with update_graph.exclusive_lock(self.test_update_graph):
+            result_table = function_generated_table(table_generator_function, refresh_interval_ms=2000,
+                                                    nrows = 5, query_string = "Timestamp = io.deephaven.base.clock.Clock.system().currentTimeMillis()")
+            self.assertEqual(result_table.size, 5)
+            first_row_key = get_row_key(0, result_table)
+            initial_time = result_table.j_table.getColumnSource("Timestamp").get(first_row_key)
+
+            if not result_table.await_update(5_000):
+                raise RuntimeError("Result table did not update within 5 seconds")
+
+            first_row_key = get_row_key(0, result_table)
+            later_time = result_table.j_table.getColumnSource("Timestamp").get(first_row_key)
+
+        # Make sure it ticked at least once within 5 seconds. It should have ticked twice,
+        # but leaving a wider margin to ensure the test passes -- as long as it ticks at all
+        # we can be confident it's working.
+        self.assertGreater(later_time, initial_time)
+
+    def test_generated_table_wrong_args(self):
+        def table_generator_function(nrows, query_string):
+            return empty_table(nrows).update(query_string)
+
+        # nrow is not a valid argument, so we expect a type error
+        with update_graph.exclusive_lock(self.test_update_graph):
+            self.assertRaises(TypeError, table_generator_function, nrow=1, query_string="Timestamp = io.deephaven.base.clock.Clock.system().currentTimeMillis()")
```

Comment:
```
       with self.assertRaises(TypeError) as cm:
            j_array = dtypes.array("java.lang.String", ["a", "b"])
        self.assertIn("expects a DType", str(cm.exception))
```
This is a pattern we use to additionally check if the error message is what we expect. In this case, we want the exception to clearly tell us the error is from calling table_generator_function not hidden inside the `function_generated_table` call.

Reply:
Recent update has corrected this.